### PR TITLE
feat: implement StakingExample

### LINKED ISSUE #1004: Add staking example
### Problem

The SDK does not have an example for the staking feature.

 - [x] https://github.com/hashgraph/hedera-sdk/issues/41

### Solution

Add example

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by rickyrodmac**
Signed-off-by: Ricky Rodriguez <ricky.rodriguez@launchbadge.com>

Created a staking example.

**Related issue(s)**:

Fixes #1004

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by danielakhterov** [comment]
@rickyrodmac  https://github.com/hashgraph/hedera-sdk/blob/main/reference/examples/StakingExample.md

File: examples/src/main/java/StakingExample.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            .accountId;
+        Objects.requireNonNull(bobId);
+
+        printAccountInfo(client, aliceId, "Alice");
+        printAccountInfo(client, bobId, "Bob");
+
+        // Alice stakes hbar to bob
+        AccountUpdateTransaction aliceTx =  new AccountUpdateTransaction()
+            .setAccountId(aliceId)
+            .setProxyAccountId(bobId);
```
Comment (SimiHunjan) [inline examples/src/main/java/StakingExample.java:85]: Proxy account ID was deprecated https://github.com/hashgraph/hedera-protobufs/pull/195. Please use node ID. This feature is available on previewnet. Please test the example works

Comment (SimiHunjan) [reply examples/src/main/java/StakingExample.java:85]: Reference [HIP-406](https://hips.hedera.com/hip/hip-406)

Comment (SimiHunjan) [reply examples/src/main/java/StakingExample.java:85]: you can use `setStakeAccountId` to stake to another account not `setProxyAccountId`

Comment (rickyrodmac) [reply examples/src/main/java/StakingExample.java:85]: I'm rewriting the example.

**Pull request merged**